codegen/funcs.go

```diff
@@ -155,3 +156,59 @@ func KebabCase(name string) string {
 	name = SnakeCase(name)
 	return strings.Replace(name, "_", "-", -1)
 }
+
+// GoTypeRef returns the Go code that refers to the Go type which matches the given data type
+func GoTypeRef(dt design.DataType) string {
+	tname := GoTypeName(dt)
+	if design.IsObject(dt) {
+		return "*" + tname
+	}
+	return tname
+}
+
+// GoTypeName returns the Go type name for a data type.
+func GoTypeName(dt design.DataType) string {
+	switch actual := dt.(type) {
+	case design.Primitive:
+		return GoNativeType(dt)
+	case *design.Array:
+		return "[]" + GoTypeRef(actual.ElemType.Type)
+	default:
+		panic(fmt.Sprintf("goa bug: unknown type %#v", actual))
+	}
+}
+
+// GoNativeType returns the Go built-in type from which instances of provided datatype can be initialized.
+func GoNativeType(t design.DataType) string {
```

Comment:
Same as above add a line to the header comment about needing to support other types.